### ISSUE TITLE
Add tags and messages to osmosdr ala uhd

### ISSUE THREAD

**Issue by xloem** [xloem/gr-osmosdr#1]
https://github.com/gnuradio/gnuradio/tree/master/gr-uhd uses tags and messages to report on metadata changes such as center frequency, and allow for a runtime control interface allowing tuning etc.

It would be great to see osmosdr adopt the same interface, so that other blocks can be made to control either one.

I'd like to see this change land in osmocom's gr-osmosdr upstream repository, which means eventually submitting a patch to the osmocom mailing list at https://lists.osmocom.org/mailman/listinfo/osmocom-sdr .

**Comment by xloem** [comment]
Note that gr-osmosdr provides a common interface to all drivers, so something like this should already exist ...

**Comment by Chaostheorie** [comment]
Is this issue still open/ the bounty active?

**Comment by xloem** [comment]
Yes.  I haven't looked at the project in a long time now, so bring my attention to things if needed.  General tags and messages will help pave a way for development of more general and powerful tools.

The bounty for closing this issue is at https://www.bountysource.com/issues/55802628-add-tags-and-messages-to-osmosdr-ala-uhd .  The payout is currently $600 .
